pipeline-model-definition/src/test/java/org/jenkinsci/plugins/pipeline/modeldefinition/OptionsTest.java

```diff
@@ -556,6 +559,24 @@ public void sameJobPropertiesNotOverride() throws Exception {
         assertSame(strategy, strategy2);
     }
 
+    @Issue("JENKINS-54250")
+    @Test
+    public void verifyDisableRestartFromStageActionIsAdded() throws Exception {
+        WorkflowRun b = expect("options/restartableFromStageDisabled").go();
+
+        DisableRestartFromStageAction action = b.getParent().getAction(DisableRestartFromStageAction.class);
+        assertNotNull(action);
+    }
+
+    @Issue("JENKINS-54250")
+    @Test
+    public void verifyDisableRestartFromStageActionIsNotAdded() throws Exception {
```

Comment:
I think it would be interesting to store the result of `expect`, keep your current assertions, but then add a `DisableRestartFromStageAction` to the parent manually and call `go` a second time to show that it gets removed.

Reply:
Please verify the change